web/packages/teleport/src/Account/SideNav.tsx

```diff
@@ -0,0 +1,181 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useHistory } from 'react-router';
+import { useLocation } from 'react-router-dom';
+import styled from 'styled-components';
+
+import cfg from 'teleport/config';
+
+import { preferencesHeadings } from './Preferences';
+import { securityHeadings } from './SecuritySettings';
+
+export interface Heading {
+  name: string;
+  id: string;
+}
+
+export type Headings = Heading[];
+
+export interface SideNavProps {
+  recoveryEnabled?: boolean;
+  trustedDevicesEnabled?: boolean;
+}
+
+export function SideNav({
+  recoveryEnabled = false,
+  trustedDevicesEnabled = false,
+}: SideNavProps) {
+  const history = useHistory();
+  const location = useLocation();
+
+  const navigateTo = (path: string) => {
+    history.replace(path);
+
+    const idToScrollTo = path.includes('#') ? path.split('#')[1] : null;
+
+    // If there's an ID found, scroll to it
+    if (idToScrollTo) {
+      // setTimeout is used to ensure the DOM is rendered before
+      // trying to scroll to it. The DOM re-renders if the user
+      // clicks on a subheading that is on a different page.
+      setTimeout(() => {
+        const element = document.getElementById(idToScrollTo);
+        if (element) {
+          element.scrollIntoView({ behavior: 'smooth' });
+        }
+      }, 0);
+    }
+  };
+
+  const navItems = generateHeadings(recoveryEnabled, trustedDevicesEnabled);
+
+  return (
+    <SideNavWrapper>
+      {navItems.map(group => {
+        // Check if this section is active based on the current path
+        const isSectionActive = location.pathname === group.page.link;
+
+        return (
+          <div key={group.page.name}>
+            <SectionTitle
+              className={isSectionActive ? 'active' : ''}
+              onClick={() => navigateTo(group.page.link)}
+            >
+              {group.page.name}
+            </SectionTitle>
+            <LinkList>
+              {group.headings.map(heading => (
+                <li key={heading.name}>
+                  <HeadingItem
+                    href={`${group.page.link}#${heading.id}`}
+                    onClick={e => {
+                      e.preventDefault();
+                      navigateTo(`${group.page.link}#${heading.id}`);
+                    }}
+                  >
+                    {heading.name}
+                  </HeadingItem>
+                </li>
+              ))}
+            </LinkList>
+          </div>
+        );
+      })}
+    </SideNavWrapper>
+  );
+}
+
+function generateHeadings(
```

Comment:
I'm not super happy with the way I'm generating headings for the side nav. It would be ideal if somehow the side nav could read all the pages (`SecuritySettings.tsx` and `Preferences.tsx` currently) to generate the headings, but not sure how that'd be done.

The current flow is the side nav calls a function in both the above pages to get their headings, and adds a couple itself if it is running enterprise edition. But if the settings are moved around on the page or more settings are added, these functions will have to be modified.